docs/tutorials/parsec-tutorial.md

```diff
@@ -0,0 +1,557 @@
+
+```sh
+*.pyc
+m5out
+.vscode
+results
+venv
+disk-image/packer
+disk-image/packer_1.4.3_linux_amd64.zip
+disk-image/parsec/parsec-image/parsec
+disk-image/parsec/
+disk-image/parsec-benchmark/
+disk-image/packer_cache
+gem5
+linux-stable/
+```
+
+gem5art relies on Python 3, so we suggest creating a virtual environment before using gem5art.
+
+```sh
+virtualenv -p python3 venv
+source venv/bin/activate
+```
+
+gem5art can be installed (if not already) using pip:
+
+```sh
+pip install gem5art-artifact gem5art-run gem5art-tasks
+```
+
+## Building gem5
+
+Next clone gem5 from googlesource:
+
+```sh
+git clone https://gem5.googlesource.com/public/gem5
+```
+Run the following build gem5:
+
+```sh
+scons build/X86/gem5.opt -j8
+```
+
+Also make sure to build the m5 utility which will be moved to the disk image eventually.
+m5 utility allows to trigger simulation tasks from inside the simulated system.
+For example, it can be used dump simulation statistics when the simulated system triggers to do so.
+We will need m5 mainly to exit the simulation when the simulated system will be done with the execution of a particular NPB benchmark.
+
+```sh
+cd gem5/util/m5/
+make -f Makefile.x86
+```
+
+## Creating a disk image
+First create a disk-image folder where we will keep all disk image related files:
+
+```sh
+mkdir disk-image
+```
+
+We will follow the similar directory structure as discussed in [Disk Images](../main-doc/disks.md) section.
+Add a folder named shared for config files which will be shared among all disk images (and will be kept to their defaults) and one folder named parsec which will contain files configured for PARSEC disk image. Add [preseed.cfg](https://github.com/darchr/gem5art/blob/master/docs/disks/shared/preseed.cfg) and [serial-getty@.service](https://github.com/darchr/gem5art/blob/master/docs/disks/shared/serial-getty@.service) in shared/.
+
+In npb/ we will add the benchmark source first, which will eventually be transferred to the disk image through our npb.json file.
+
+```sh
+cd disk-image/npb
+git clone https://github.com/darchr/npb-hooks.git
+```
+
+This source of NPB has ROI (region of interest) annotations for each benchmark which will be used by gem5 to
+separate out simulation statistics of the important parts of a program from the rest of the program.
+Basically, gem5 magic instructions are used before and after the ROI which exit the guest and transfer control to gem5 run script which can then do things like dumping or resetting stats or switching to cpu of interest.
+
+Next, we will add few other files in npb/ which will be used for compilation of NPB inside the disk image and eventually running of these benchmarks with gem5.
+These files will be moved from host to the disk image using npb.json file as we will soon see.
+
+First, create a file npb-install.sh, which will be executed inside the disk image (once it is built) and will install NPB on the disk image:
```

Comment:
Ok.. I'm going to stop saying "parsec" over and over... all of this needs to be updated.